File: Easy/0190_reverse-bits/reverse-bits.go
package main

import (
	"fmt"
)

func reverseBits(n int) int {
  rev := 0
	for i := 0; i < 32; i++ {
		rev = (rev << 1) | (n & 1)
		n >>= 1
	}
	return rev
}

func main() {
	tests := []int{
		43261596,
		2147483644,
		0,
		1,
	}

	for _, n := range tests {
		fmt.Printf("---\nN, %d", n)
		fmt.Printf("reverseBits: %d\n", reverseBits(n))
	}
}

File: Easy/0190_reverse-bits/reverse-bits.py
#!/bin/python3

from typing import List

class Solution:
    def reverseBits(self, n: int) -> int:
        rev : int = 0
        for i in range(32):
            rev = (rev << 1) | (n & 1)
            n >>= 1
        return rev

def main():
    tests : List[int] = [
        43261596,
        2147483644,
        0,
        1,
    ]
    for n in tests:
        print(f"---\nN: {n}\nreverseBits: {Solution().reverseBits(n)}")

if __name__ == "__main__":
    main()
